conjur/resource.py

```diff
@@ -28,3 +32,74 @@ def __init__(self, api, kind, identifier):
     @property
     def resourceid(self):
         return ":".join([self.api.config.account, self.kind, self.identifier])
+
+    def permit(self, role, privilege, grant_option=False):
+        '''
+        Permit `role` to perform `privilege` on this resource.  If
+        `grant_option` is True, the role will be able to grant this
+        permission to other resources.
+
+        You must own the resource or have the permission with `grant_option`
+        to call this method.
+        '''
+        data = {}
+        params = {
+            'permit': 'true',
+            'privilege': privilege,
+            'role': authzid(role, 'role')
+        }
+        if grant_option:
+            data['grant_option'] = 'true'
+
+        self.api.post(self.url(), data=data, params=params)
+
+    def deny(self, role, privilege):
+        '''
+        Deny `role` permission to perform `privilege` on this resource.
+
+        You must own the resource or have the permission with `grant_option`
+        on it to call this method.
+        '''
+        params = {
+            'permit': 'true',
+            'privilege': privilege,
+            'role': authzid(role)
+        }
+
+        self.api.post(self.url(), parmas=params)
+
+    def check_permission(self, privilege, role=None):
+        '''
+        Return True if +role+ has +privilege+ on this resource.
+
+        +role+ may be a Role instance, an object with a +role+ method,
+        or a role id as a string.
+
+        If +role+ is not given, check the permission for the currently
+        authenticated role.
+        '''
+
+        if role is None:
+            # Handle self role check
+            response = self.api.get(self.url(),
+                                    params={'check': 'true',
+                                            'privilege': privilege},
+                                    check_errors=False)
+            if response.status_code == 204:
+                return True
+            elif response.status_code in (404, 403, 409):
```

Comment:
Yeah I was a little confused about that too.  I mean, I don't know why this would ever raise `409 conflict` since it's not modifying anything.   I suspect the Ruby API is doing the right thing and 409 is a copy/paste error in apiary.  I'll cut this back to 404 and 403.